streams/readable-byte-streams/bad-buffers-and-views.any.js

```diff
@@ -203,6 +203,38 @@ async_test(t => {
 }, 'ReadableStream with byte source: respondWithNewView() throws if the supplied view is zero-length on a ' +
     'non-zero-length buffer (in the readable state)');
 
+async_test(t => {
+  const stream = new ReadableStream({
+    pull: t.step_func_done(c => {
+      const view = c.byobRequest.view.subarray(1, 2);
+
+      assert_throws_js(RangeError, () => c.byobRequest.respondWithNewView(view));
+    }),
+    type: 'bytes'
+  });
+  const reader = stream.getReader({ mode: 'byob' });
+
+  reader.read(new Uint8Array([4, 5, 6]));
+}, 'ReadableStream with byte source: respondWithNewView() throws if the supplied view has a different offset ' +
+   '(in the readable state)');
```

Comment:
Drive-by fix: I noticed that [this check](https://github.com/whatwg/streams/blob/53282dde15be5f844768c2167405660da3c29b2d/reference-implementation/lib/abstract-ops/readable-streams.js#L1657-L1659) was not covered by any tests, so I added one. 🙂